scripts/spreadsheet-import/index.js

```diff
@@ -0,0 +1,285 @@
+const {promisify} = require('util');
+const {getSheetData} = require('./spreadsheet-api');
+const {processSheet, simplifySpreadsheetData} = require('./spreadsheet-utils');
+const rimraf = promisify(require('rimraf'));
+const fetch = require('node-fetch');
+const imageType = require('image-type');
+const imageSize = require('image-size');
+const mkdirp = require('mkdirp');
+
+const timeout = promisify(setTimeout);
+
+// spreadsheet-format is illustrated here:
+//   https://docs.google.com/spreadsheets/d/14TQHTYePS0SAaXGRNF3zYXvvk8xz25CXW-uekQy4HAs/edit
+
+program
+  .description(
+    'import speaker- and talk-data from the specified spreadheet and ' +
+      'update the files in contents/speakers and contents/talks'
+  )
+  .arguments('<spreadsheet>')
+  .action(spreadsheet => {
+    const rxSpreadsheetIdFromUrl = /^https:\/\/docs\.google\.com\/.*\/d\/([^/]+).*$/;
+
+    program.spreadsheetId = spreadsheet;
+
+    if (rxSpreadsheetIdFromUrl.test(spreadsheet)) {
+      program.spreadsheetId = spreadsheet.replace(rxSpreadsheetIdFromUrl, '$1');
+    }
+  })
+  .option(
+    '-p --production',
+    "run in production-mode (don't import unpublished items)"
+  )
+  .option('-i --image-path <imagePath>', 'alternative path to look for images')
+  .option('-C --no-cleanup', "don't run cleanup before import")
+  .parse(process.argv);
+
+const contentRoot = path.resolve(__dirname, '../../contents');
+const sheetParams = {
+  speakers: {
+    templateGlobals: {
+      template: 'pages/speaker.html.njk'
+    },
+    dataFieldName: 'speaker',
+    contentPath: 'speakers'
+  },
+  talks: {
+    templateGlobals: {
+      template: 'pages/talk.html.njk'
+    },
+    dataFieldName: 'talk',
+    contentPath: 'talks'
+  }
+};
+
+const wwwtfrcFile = __dirname + '/../../.wwwtfrc';
+const hasRcFile = fs.existsSync(wwwtfrcFile);
+
+let rcFileParams = {};
+if (hasRcFile) {
+  rcFileParams = JSON.parse(fs.readFileSync(wwwtfrcFile));
+}
+
+const params = {
+  ...rcFileParams,
+  imagePath: program.imagePath,
+  doCleanup: program.cleanup,
+  publishedOnly: program.production || process.env.NODE_ENV === 'production'
+};
+if (program.spreadsheetId) {
+  params.spreadsheetId = program.spreadsheetId;
+}
+
+if (!params.spreadsheetId) {
+  console.log(
+    chalk.red.bold('A spreadsheet-id (or spreadsheet-url) is required.')
+  );
+  program.outputHelp();
+  process.exit(1);
+}
+
+if (!hasRcFile) {
+  console.log('saving settings to', chalk.green('.wwwtfrc'));
+  fs.writeFileSync(
+    wwwtfrcFile,
+    JSON.stringify({spreadsheetId: params.spreadsheetId}, null, 2)
+  );
+}
+
+main(params).catch(err => console.error(err));
+
+async function main(params) {
+  // ---- ensure the directories exist...
+  const requiredDirectories = [
+    'speakers',
+    'talks',
+    'images/speaker',
+  ];
+  const requiredDirectoryPaths = requiredDirectories.map(dir => `${__dirname}/../../contents/${dir}`);
+
+  if (requiredDirectoryPaths.some(dirPath => !fs.existsSync(dirPath))) {
```

Comment:
I probably could have used `filter` and call `mkdirp` only on the directories that actually need to be created… WDYT?

```js
const missingDirectories = requiredDirectoryPaths.filter(dir => !fs.existsSync(dirPath));

if (missingDirectories.length) {
  console.log(chalk.gray('creating missing directories...'));
  missingDirectories.forEach(dir => mkdirp(dir));
}
```